Framework/API/src/Algorithm.cpp

```diff
@@ -959,8 +959,9 @@ IAlgorithm_sptr Algorithm::fromHistory(const AlgorithmHistory &history) {
  */
 IAlgorithm_sptr Algorithm::fromString(const std::string &input) {
   ::Json::Value root;
-  ::Json::Reader reader;
-  if (reader.parse(input, root)) {
+  ::Json::CharReaderBuilder readerBuilder;
+  auto reader = std::unique_ptr<::Json::CharReader>(readerBuilder.newCharReader());
```

Comment:
Naive question. Can we not just instantiate the `::Json::CharReader` class directly on the stack without the need to new?

Reply:
To create a CharReader using a CharReaderBuilder it returns a pointer. https://open-source-parsers.github.io/jsoncpp-docs/doxygen/class_json_1_1_char_reader_builder.html#a81da7da750111321ff14baf0f0a4c8ae